gst/nnstreamer/tensor_common.c

```diff
@@ -1132,17 +1133,21 @@ gst_tensor_set_current_time (GstCollectPads * collect,
       return isEOS;
     }
 
-    switch (sync) {
+    switch (sync.mode) {
       case SYNC_SLOWEST:
         if (*current_time < GST_BUFFER_PTS (buf))
           *current_time = GST_BUFFER_PTS (buf);
         gst_buffer_unref (buf);
         break;
       case SYNC_BASEPAD:
+        if (count == sync.data_basepad.sink_id)
+          *current_time = GST_BUFFER_PTS (buf);
+        gst_buffer_unref (buf);
         break;
       default:
         break;
     }
+    count++;
```

Comment:
unref-buffer
remove line 1140 and 1145, and add here 'gst_buffer_unref (buf);'

Reply:
right. I will change it.